src/backend/api/meals.js

```diff
@@ -1,15 +1,97 @@
+const { response, request } = require("express");
 const express = require("express");
 const router = express.Router();
 const knex = require("../database");
 
 router.get("/", async (request, response) => {
   try {
-    // knex syntax for selecting things. Look up the documentation for knex for further info
+
+    
+    if('maxPrice' in request.query){
+      const maxPrice = request.query.maxPrice;
+      const highPriceMeals = await knex ('meal').where ('price', '<=', maxPrice);
+      response.send(highPriceMeals);
+      return;
+    }
+    if('availableReservations' in request.query){
+      const availableReservations = (request.query.availableReservations == 'true');
+      const numberOfReservations = await knex.from('meal').join('reservations','meal.id', '=', 'reservations.id')
+      .select('meal.title')
+      .where(('max_reservations' - 'reservations.number_of_guests', true), availableReservations)
+      response.send(numberOfReservations);
+      return;
+    }
+    if('title' in request.query){
+      const title = request.query.title;
+      const mealsTitle = await knex('meal').where('title', 'like', '%' + title + '%');
+      response.send(mealsTitle);
+      return;
+    }
+    if('createdAfter' in request.query){
+      const createdAfter = request.query.createdAfter;
```

Comment:
```suggestion
      const createdAfter = Date.parse(request.query.createdAfter);
```
be carefull with querry params . you need to check if the data is correct

Reply:
Okay ..Thank you !!